nixos/modules/services/misc/preload.nix

```diff
@@ -0,0 +1,40 @@
+# flatpak service.
+{ self, config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.services.preload;
+  preload = pkgs.preload;
```

Comment:
What about adding a `package` option using `mkPackageOptionMD` and reference that package using `cfg.services.preload.package`?